drivers/display/display_st7567.c

```diff
@@ -206,21 +206,37 @@ static int st7567_suspend(const struct device *dev)
 }
 
 static int st7567_write_default(const struct device *dev, const uint16_t x, const uint16_t y,
-				const struct display_buffer_descriptor *desc, const void *buf,
-				const size_t buf_len)
+				const uint8_t *buf, const size_t buf_len)
 {
+	const struct st7567_config *config = dev->config;
 	int ret;
 	uint8_t cmd_buf[3];
+	uint16_t column = x + config->column_offset;
+
+	cmd_buf[0] = ST7567_COLUMN_LSB | (column & 0xF);
+	cmd_buf[1] = ST7567_COLUMN_MSB | ((column >> 4) & 0xF);
+	cmd_buf[2] = ST7567_PAGE | (y >> 3);
+	ret = st7567_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+	if (ret < 0) {
+		return ret;
+	}
+	ret = st7567_write_bus(dev, (uint8_t *)buf, buf_len, false);
```

Comment:
Could be cast to const uint8_t* instead in both places sonarqube reports but that was already incorrect before so as you want.